q2_types/per_sample_sequences/_format.py

```diff
@@ -137,21 +142,56 @@ def sequences_path_maker(self, sample_id, barcode_id, lane_number,
         return '%s_%s_L%03d_R%d_001.fastq.gz' % (sample_id, barcode_id,
                                                  lane_number, read_number)
 
+    def _find_duplicates(self, ids):
+        return {x for x, c in collections.Counter(ids).items() if c > 1}
+
+    def validate(self):
+        forwards = []
+        reverse = []
+        for p in self.path.iterdir():
+            if p.is_dir():
+                d = p.relative_to(self.path)
+                raise ValidationError("Contains a subdirectory: %s" % d)
+            else:
+                if p.name.endswith('_001.fastq.gz'):
+                    sample_id = p.name.rsplit('_', maxsplit=4)[0]
+                    if p.name.endswith('R1_001.fastq.gz'):
+                        forwards.append(sample_id)
+                    else:
```

Comment:
Should we require that reverse reads end with `R2_001.fastq.gz` or are other parts of the codebase more lenient in this regard?

Reply:
Technically it must be `R2_001` at this point, as all of the File/FileCollections have validated, and there aren't any extra files unaccounted for at this point and the regex requires either R1 or R2.